webapp/docker/entrypoint.sh

```
@@ -0,0 +1,103 @@
+  local -a all_vars
+  mapfile -t all_vars < <(extract_env_vars)
+  
+  if [[ ${#all_vars[@]} -eq 0 ]]; then
+    log "ERROR: No environment variables found in $ENV_TS"
+    exit 1
+  fi
+  log "Found ${#all_vars[@]} vars: ${all_vars[*]}"
+
+  # Auto-generate deployment timestamp for preview environments
+  if [[ -n "${GIT_BRANCH:-}" && -z "${DEPLOYED_AT:-}" ]]; then
+    export DEPLOYED_AT=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)